From c5fd289209bdf5329d1b15d211f909b642a9a8ce Mon Sep 17 00:00:00 2001
From: Benjamin Tissoires <benjamin.tissoires@redhat.com>
Date: Wed, 23 Jul 2014 13:27:06 -0400
Subject: [input] synaptics: add min/max quirk for the ThinkPad W540

Message-id: <1406122030-5632-11-git-send-email-benjamin.tissoires@redhat.com>
Patchwork-id: 86353
O-Subject: [RHEL7.1 / RHEL7.0.z PATCH 10/14] Input: synaptics - add min/max quirk for the ThinkPad W540
Bugzilla: 1093449
RH-Acked-by: Aristeu S. Rozanski F. <arozansk@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Peter Hutterer <peter.hutterer@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1093449
Upstream: since v3.15, backported in stable
Brew: https://brewweb.devel.redhat.com/taskinfo?taskID=7722437
Test: tested by myself on a t440s and a t410, was extensively tested in RHEL 6.6

commit 0b5fe736fe923f1f5e05413878d5990e92ffbdf5
Author: Hans de Goede <hdegoede@redhat.com>
Date:   Wed May 14 11:10:40 2014 -0700

    Input: synaptics - add min/max quirk for the ThinkPad W540

    https://bugzilla.redhat.com/show_bug.cgi?id=1096436

    Cc: stable@vger.kernel.org
    Tested-and-reported-by: ajayr@bigfoot.com
    Signed-off-by: Hans de Goede <hdegoede@redhat.com>
    Signed-off-by: Dmitry Torokhov <dmitry.torokhov@gmail.com>

Signed-off-by: Benjamin Tissoires <benjamin.tissoires@redhat.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/input/mouse/synaptics.c b/drivers/input/mouse/synaptics.c
index d68d33f..d4c05b1 100644
--- a/drivers/input/mouse/synaptics.c
+++ b/drivers/input/mouse/synaptics.c
@@ -1614,6 +1614,14 @@ static const struct dmi_system_id min_max_dmi_table[] __initconst = {
   .driver_data = (int []){1024, 5112, 2024, 4832},
  },
  {
+  /* Lenovo ThinkPad W540 */
+  .matches = {
+   DMI_MATCH(DMI_SYS_VENDOR, "LENOVO"),
+   DMI_MATCH(DMI_PRODUCT_VERSION, "ThinkPad W540"),
+  },
+  .driver_data = (int []){1024, 5112, 2024, 4832},
+ },
+ {
   /* Lenovo Yoga S1 */
   .matches = {
    DMI_MATCH(DMI_SYS_VENDOR, "LENOVO"),
-- 
1.7.1